src/components/SearchField.tsx

```diff
@@ -46,10 +48,11 @@ const SearchField: React.FC = () => {
   };
 
   const submitSearchRequest = (event: React.SyntheticEvent) => {
+    setDropdown(false);
     event.preventDefault();
     if (inputValue.trim()) {
       history.push(
-        `/search-results/${inputValue}/${genreToSearch}/${dateRange.fromYear}/${dateRange.toYear}/${checkboxes.favorites}/${checkboxes.history}/1`
+        `/search-results?title=${inputValue}&genre=${genreToSearch}&fromYear=${dateRange.fromYear}&toYear=${dateRange.toYear}&favorites=${checkboxes.favorites}&history=${checkboxes.history}&page=1`
```

Comment:
Please use search param for query, and create query via URLSearchParams().
Also avoid setting undefined values there - it doesent looks nice

Reply:
I've refactored it but don't know if it's good.